diffsync/store/redis.py

```diff
@@ -0,0 +1,242 @@
+        uid = obj.get_unique_id()
+
+        # Get existing Object
+        object_key = self._get_key_for_object(modelname, uid)
+
+        existing_obj_binary = self._store.get(object_key)
+        if existing_obj_binary:
+            existing_obj = loads(existing_obj_binary)
+            existing_obj_dict = existing_obj.dict()
+
+            if existing_obj_dict != obj.dict():
+                raise ObjectAlreadyExists(f"Object {uid} already present", obj)
+
+            # Return so we don't have to change anything on the existing object and underlying data
+            return
+
+        # Remove the diffsync object before sending to Redis
+        obj_copy = copy.copy(obj)
+        obj_copy.diffsync = None
+
+        self._store.set(object_key, dumps(obj_copy))
+
+    def update(self, obj: "DiffSyncModel"):
+        """Update a DiffSyncModel object to the store.
+
+        Args:
+            obj (DiffSyncModel): Object to update
+        """
+        modelname = obj.get_type()
+        uid = obj.get_unique_id()
+
+        object_key = self._get_key_for_object(modelname, uid)
+        obj_copy = copy.copy(obj)
+        obj_copy.diffsync = None
+
+        self._store.set(object_key, dumps(obj_copy))
+
+    def remove(self, obj: "DiffSyncModel", remove_children: bool = False):
+        """Remove a DiffSyncModel object from the store.
+
+        Args:
+            obj (DiffSyncModel): object to remove
+            remove_children (bool): If True, also recursively remove any children of this object
+
+        Raises:
+            ObjectNotFound: if the object is not present
+        """
+        modelname = obj.get_type()
+        uid = obj.get_unique_id()
+
+        object_key = self._get_key_for_object(modelname, uid)
+
+        if not self._store.exists(object_key):
+            raise ObjectNotFound(f"{modelname} {uid} not present in Cache")
+
+        if obj.diffsync:
+            obj.diffsync = None
+
+        self._store.delete(object_key)
+
+        if remove_children:
+            for child_type, child_fieldname in obj.get_children_mapping().items():
+                for child_id in getattr(obj, child_fieldname):
+                    try:
+                        child_obj = self.get(child_type, child_id)
+                        self.remove(child_obj, remove_children=remove_children)
+                    except ObjectNotFound:
+                        pass
+                        # Since this is "cleanup" code, log an error and continue, instead of letting the exception raise
+                        # self._log.error(f"Unable to remove child {child_id} of {modelname} {uid} - not found!")
+
+    def count(self, modelname=None):
+        """Returns the number of elements of an specific model name."""
+        search_pattern = f"{self._store_label}:*"
+        if modelname:
+            search_pattern = f"{self._store_label}:{modelname.lower()}:*"
+
+        return sum([1 for _ in self._store.scan_iter(search_pattern)])
```

Comment:
Can we not do something like:

```suggestion
        return len(self._store.scan_iter(search_pattern))
```